src/vm.rb

```diff
@@ -75,19 +71,32 @@ def running?
 
     def state
       case domain.info.state
-      when 5
-        'off'
+      when 0
+        'no state'
       when 1
         'on'
+      when 2
+        'blocked'
+      when 3
+        'paused'
+      when 4
+        'shutting down'
+      when 5
+        'off'
+      when 6
+        'crashed'
+      when 7
+        'suspended'
```

Comment:
Could be worth adding an `else` which raises here? Just in case a new state appears in the future so we get an error rather than this just returning `nil`.